src/main/java/com/hazelcast/gcp/RestClient.java

```diff
@@ -117,6 +118,15 @@ private static String read(InputStream stream) {
         return scanner.next();
     }
 
+    private void handleErrorStream(String errorStream, String method) {
+        if (GcpConnectionException.getIsGcpConnectionException(errorStream)) {
```

Comment:
I'd avoid using some GCP specific classes, in the general class like `RestClient`.